deepchem/feat/molecule_featurizers/dmpnn_featurizer.py

```diff
@@ -0,0 +1,157 @@
+  # + 2 at end for is_in_aromatic and mass
+
+
+def get_atomic_num_one_hot(atom: RDKitAtom,
+                           allowable_set: List[int],
+                           include_unknown_set: bool = True) -> List[float]:
+  """Get a one-hot feature about atomic number of the given atom.
+
+  Parameters
+  ---------
+  atom: rdkit.Chem.rdchem.Atom
+    RDKit atom object
+  allowable_set: List[int]
+    The range of atomic numbers to consider.
+  include_unknown_set: bool, default False
+    If true, the index of all types not in `allowable_set` is `len(allowable_set)`.
+
+  Returns
+  -------
+  List[float]
+    A one-hot vector of atomic number of the given atom.
+    If `include_unknown_set` is False, the length is `len(allowable_set)`.
+    If `include_unknown_set` is True, the length is `len(allowable_set) + 1`.
+  """
+  return one_hot_encode(atom.GetAtomicNum() - 1, allowable_set,
+                        include_unknown_set)
+
+
+def get_atom_chiral_tag_one_hot(
+    atom: RDKitAtom,
+    allowable_set: List[int],
+    include_unknown_set: bool = True) -> List[float]:
+  """Get a one-hot feature about chirality of the given atom.
+
+  Parameters
+  ---------
+  atom: rdkit.Chem.rdchem.Atom
+    RDKit atom object
+  allowable_set: List[int]
+    The list of chirality tags to consider.
+  include_unknown_set: bool, default False
+    If true, the index of all types not in `allowable_set` is `len(allowable_set)`.
+
+  Returns
+  -------
+  List[float]
+    A one-hot vector of chirality of the given atom.
+    If `include_unknown_set` is False, the length is `len(allowable_set)`.
+    If `include_unknown_set` is True, the length is `len(allowable_set) + 1`.
+  """
+  return one_hot_encode(atom.GetChiralTag(), allowable_set, include_unknown_set)
+
+
+def get_atom_mass(atom: RDKitAtom) -> List[float]:
+  """Get vector feature containing downscaled mass of the given atom.
+
+  Parameters
+  ---------
+  atom: rdkit.Chem.rdchem.Atom
+    RDKit atom object
+
+  Returns
+  -------
+  List[float]
+    A vector of downscaled mass of the given atom.
+  """
+  return [atom.GetMass() * 0.01]
+
+
+def atom_features(
+    atom: Chem.rdchem.Atom,
+    functional_groups: List[int] = None) -> List[Union[bool, int, float]]:
+  """Helper method used to compute atom feature vector.
+
+  Deepchem already contains an atom_features function, however we are defining a new one here due to the need to handle features specific to DMPNN.
+
+  Parameters
+  ----------
+  atom: RDKit.Chem.rdchem.Atom
+    Atom to compute features on.
+  functional_groups: List[int]
+    A k-hot vector indicating the functional groups the atom belongs to.
+    Default value is None
+
+  Returns
+  -------
+  features: List[Union[bool, int, float]]
+    An list of atom features.
+
+  Examples
+  --------
+  >>> from rdkit import Chem
+  >>> mol = Chem.MolFromSmiles('C')
+  >>> atom = mol.GetAtoms()[0]
+  >>> features = dc.feat.molecule_featurizers.dmpnn_featurizer.atom_features(atom)
```

Comment:
You need to create an actual DeepChem featurizer object that inherits from `MolecularFeaturizer`. See for example https://deepchem.readthedocs.io/en/latest/api_reference/featurizers.html#deepchem.feat.ConvMolFeaturizer. Are you planning to do this in a future PR?

Reply:
Ok reading your comment I think you are planning to do this in a future PR if I understand correctly

Reply:
I plan to implement featurizer support functions, and graph object class for this week and will try to implement the DMPNNFeaturizer class and DMPNNEncoding class next week.